llvm/tools/objwriter/debugInfo/dwarf/dwarfGen.cpp

```diff
@@ -492,96 +492,31 @@ static void EmitVarLocation(MCObjectStreamer *Streamer,
 class LexicalScope
 {
 public:
-  LexicalScope(uint64_t Start, uint64_t End, bool IsFuncScope = false) :
-               Start(Start),
-               End(End),
-               IsFuncScope(IsFuncScope) {}
-
-  LexicalScope(VarInfo *Info) :
-               Start(Info->GetStartOffset()),
-               End(Info->GetEndOffset()),
-               IsFuncScope(false) { Vars.push_back(Info); }
-
-  bool IsContains(const VarInfo *Info) const {
-    return Start <= Info->GetStartOffset() && End >= Info->GetEndOffset();
-  }
+  /// At the moment, the lexical scope reflects IL, not C#, meaning that
+  /// there is only one scope for the whole method. We could be more precise
+  /// in the future by pulling the scope information from the PDB.
```

Comment:
Do you know how the lexical scope information is used by debugger? I suppose it allows the debugger to provide the values of only the subset of locals that are relevant in the current context of the function?

Note that the JIT will shorten and lengthen the live ranges of locals in arbitrary ways when optimizing, but presumably the debugger handles the live ranges and the lexical scopes as two separate sources of "which local values should be displayed here".

In debug the JIT will extend the live range of all IL locals to the full function, regardless of when they are last used (both as a throughput optimization and for diagnostic purposes). Without having these lexical scopes I assume it means that the debugger may display multiple separate locals of the same name (e.g. think two `i` induction variables from separate loops). If I understand correctly this is what makes having these lexical scopes desirable.

Reply:
Yes that description sounds right to me.